### PR TITLE
Ts 10

### PR DESCRIPTION
**Pull request by ahammednafih**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -16,11 +16,13 @@ class User < ActiveRecord::Base
   validates_confirmation_of :password
   validates_presence_of :password_digest
 
-  before_create :confirmation_token
-  before_save :full_name
+  before_create :set_confirmation_token
+  before_save :set_full_name
+  after_update :send_reset_token_email, :if => :reset_password_token_changed?
+  after_create :send_confirmation_token_email
```
Comment (Navinnair) [inline app/models/user.rb:22]: send_confirmation_mail , try to make names minimal and understandable

File: app/models/user.rb [hunk 2]
```diff
@@ -16,11 +16,13 @@ class User < ActiveRecord::Base
...
 
-  def full_name                                                                                                                                                                                    
-    self.full_name = ([self.first_name, self.last_name] - ['']).compact.join(' ')                         
+  def set_full_name                                                                                                                                                                                    
+    self.full_name = ([self.first_name, self.last_name] - ['']).compact.join(' ')
```
Comment (Navinnair) [inline app/models/user.rb:25]: read the style guide. and change this block

File: app/models/user.rb [hunk 3]
```diff
@@ -33,15 +35,34 @@ def password_confirmation=(plaintxt_pass)
   end
 
   def authenticate(plaintxt_pass)
-    if BCrypt::Password.new(self.password_digest) == plaintxt_pass
+    if BCrypt::Password.new(self.password_digest) == plaintxt_pass && self.email_confirmed
```
Comment (Navinnair) [inline app/models/user.rb:38]: and brackets to increase readability

File: app/models/user.rb [hunk 4]
```diff
@@ -33,15 +35,34 @@ def password_confirmation=(plaintxt_pass)
...
+
+  def set_email_confirmed
+    email_confirmed = true
+    confirm_token = nil
+    self.update_attributes(:email_confirmed => email_confirmed, :confirm_token => confirm_token)
+  end
   private
-    def confirmation_token
+    def set_confirmation_token
       if self.confirm_token.blank?
```
Comment (Navinnair) [inline app/models/user.rb:56]: instead of blank try to user present?

File: app/models/user.rb [hunk 5]
```diff
@@ -33,15 +35,34 @@ def password_confirmation=(plaintxt_pass)
...
           self.confirm_token = Services::UserServices.generate_token
       end
     end
+
+    def send_reset_token_email
```
Comment (Navinnair) [inline app/models/user.rb:61]: names can be changed to send_reset_token like this

**Review by Navinnair — CHANGES_REQUESTED** [review]
please go through the review

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -1,15 +1,13 @@
 class SessionsController < ApplicationController
   skip_before_filter :authorized, :only => [:new, :create]
+  before_filter :check_login, :only =>[:new]
```
[the remark this answers is not in the document]
Comment (anbk77) [reply app/controllers/sessions_controller.rb:3]: Its used to redirect user if logged in from login page to notes path

**Pull request merged**